pkg/querier/queryrange/downstreamer.go

```diff
@@ -18,7 +18,7 @@ import (
 )
 
 const (
-	DefaultDownstreamConcurrency = 32
+	DefaultDownstreamConcurrency = 128
```

Comment:
Q: Failing to understand why it got increased (even after reading below comment on `Downstreamer()`)

Reply:
Or may be cap it higher ?

Reply:
This was historically used to limit how many downstream queries (query-frontend -> querier) could be dispatched in parallel by a single time splitted logql query. Now that this part is controlled by the `LimitedRoundTripper` instead, we only want to use the `Downstreamer` concurrency to prevent us from creating unbounded goroutines. Increasing the limit to 128 still seems reasonable to that effect but is also high enough to not pre-limit anything the `LimitedRoundTripper` would limit anyway. Basically, this is a crude attempt to prevent us from blowing up goroutines due to malicious queries without introducing a bottleneck in our query path.

Reply:
Thanks for clarifying Owen. :+1: 

Two things:
1. Can we add this 
```
 Increasing the limit to 128 still seems reasonable to that effect but is also high enough to not pre-limit anything the LimitedRoundTripper would limit anyway. 
```
as a comment to the const itself?

2. Also +1 to have this value same as `max_query_parallelism`. Because having `max_query_parallelism` as 256 will be allowed happily by `LimitedRoundTripper` but still can get limited by this `Downstreamer`.?

Reply:
> Any way we can line it up with the max_query_parallelism ? Because 128 is half the biggest tenant we have at 256 so he will get limited by this.
Yes, we could, but this is also post-splitted code, meaning that each split would need to schedule more than 128 queries for this to limit it. We may ultimately want to thread it into the `MaxQueryParallelism` code, but I felt that was overengineering for the moment and could be done in another PR if needed.

Reply:
Sounds good !